src/sync/remote_mongo_database.hpp

```diff
@@ -0,0 +1,69 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef CORE_REMOTE_MONGO_DATABASE_HPP
+#define CORE_REMOTE_MONGO_DATABASE_HPP
+
+#include <string>
+#include <json.hpp>
+#include "remote_mongo_collection.hpp"
+
+namespace realm {
+namespace app {
+
+class StitchServiceClient;
+
+class RemoteMongoDatabase {
+    
+public:
+
+    using Document = nlohmann::json;
+
+    /// The name of this database
+    const std::string name;
+    
+    RemoteMongoDatabase(std::string name, StitchServiceClient service, RemoteMongoClient client) :
```

Comment:
[minor] Service client and RemoteMongoClient should be passed by reference.

Reply:
Addressed